packages/flutter/lib/src/painting/superellipse_shape.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'basic_types.dart';
+import 'border_radius.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// Creates a shape constituting a transition between a rectangle and a circle.
```

Comment:
It would be nice if we explained how this is different from an `RRect`

Reply:
Yes - but is there any detail we can add to explain how this is different from an RRect without sending people off to wikipedia?

Reply:
Good question... 

One of the key differences is that its harder to spot where the curve on the corner ends compared to a regular rectangle with rounded corners... is it too much of a mouthful to write something like that?

Reply:
I mean it is an RRect, it's just that the radius of the corners are calculated in a way that makes them smoother than regular rounded corners i guess.

Reply:
Yeah that makes it a bit easier to understand, thanks :)